sandpack-react/src/components/CodeEditor/CodeMirror.tsx

```diff
@@ -389,16 +389,23 @@ export const CodeMirror = React.forwardRef<CodeMirrorRef, CodeMirrorProps>(
     React.useEffect(() => {
       if (cmView.current && code !== internalCode) {
         const view = cmView.current;
-
-        const selection = view.state.selection.ranges.some(
-          ({ to, from }) => to > code.length || from > code.length
-        )
-          ? EditorSelection.cursor(code.length)
-          : view.state.selection;
-
-        const changes = { from: 0, to: view.state.doc.length, insert: code };
-
-        view.dispatch({ changes, selection });
+        const selection =
+          view.state.selection.ranges.some(
+            ({ to, from }) => to > code?.length || from > code?.length
+          ) && code !== undefined
```

Comment:
This additional check means that when the only active file is deleted, there is no leftover code in the codeEditor.

Reply:
iirc I attempted this and it caused errors. I like your idea of falling back to the entry file and will attempt that first. Cheers for the feedback!